pkg/experimental/tests/rpm/source_date_epoch/rpm_contents_vs_manifest_test.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import csv
+import io
+import os
+import rpm_util
+from rules_python.python.runfiles import runfiles
+
+# Tue Mar 23 00:00:00 EDT 2021
+EXPECTED_EPOCH = '1616472000'
+EXPECTED_RPM_MANIFEST_CSV = """
+path,mtime
+/test_dir/a,{epoch}
+/test_dir/b,{epoch}
+""".strip().format(epoch=EXPECTED_EPOCH)
+
+
+def version_to_string(version):
+    return ".".join(str(i) for i in version)
+
+
+class PkgRpmCompManifest(unittest.TestCase):
+    # Support for SOURCE_DATE_EPOCH is only available as of rpm
+    # 4.14: http://rpm.org/wiki/Releases/4.14.0
+    #
+    # TODO(nacl): it would probably be better to ask the rpmbuild(8) binary for
```

Comment:
I wonder if.....
The rpmbuild toolchain can add the version as a property (or just a field saying "supports_source_date_epoch")
We can have an rpm_test rule which is just a py/sh_test that depends on the toolchain
The test rule can pass the version or support indicator to the test runner. 

On the one hand, that feels more precise than running RPM build to find out if the test can run.
But... it seems like so much more scaffolding than needed.

Having the toolchain provide the data and the test depend on it. Great.
Passing the data into the test runner. Ugly.

Sort of need incompatible target skipping for tests, which maybe exists.

Reply:
> I wonder if.....
> The rpmbuild toolchain can add the version as a property (or just a field saying "supports_source_date_epoch")
> We can have an rpm_test rule which is just a py/sh_test that depends on the toolchain
> The test rule can pass the version or support indicator to the test runner.

Hm.  I tried writing a wrapper around python tests similar when I was working on testing TreeArtifact manipulations.  Couldn't quickly determine how to create something that looks like a custom `py_test` from a rule, so I gave up and wrote a macro instead.

The macro approach could be taken here too.  Doesn't seem like it would be too hard to get something saying "supports_source_date_epoch" into a `config_setting`.

> On the one hand, that feels more precise than running RPM build to find out if the test can run.
> But... it seems like so much more scaffolding than needed.

Yup, that was my thought.  There's way too much going on here.

What is present here seems "good enough" for now until we can easily arbitrary versions of rpmbuild with bazel.

> Having the toolchain provide the data and the test depend on it. Great.
> Passing the data into the test runner. Ugly.

Indeed

> Sort of need incompatible target skipping for tests, which maybe exists.

This is what I want to see happen, but it just looks to be highly complex for little gain.

I have with rules_pkg's testing system for things like debian packages -- if the `dpkg` tools are missing from the system, then the tests are skipped within

Reply:
SGTM